plonky2x/src/frontend/eth/beacon/builder.rs

```diff
@@ -40,7 +45,36 @@ impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
         self.add_simple_generator(&generator);
         generator.validator
     }
+
+    /// SSZ Merkle Proof 
```

Comment:
nit: use full sentences for function comments. Maybe something like

`Computes the expected merkle root given a leaf, branch, and deterministic index.`